src/components/Paint.tsx

```diff
@@ -482,7 +482,9 @@ function Paint(props: PaintProps) {
                         touchPos.current = lastTouchPos;
                     }}
                     onWheel={e => {
-                        // TODO: Use e.deltaY to zoom into the canvas?
+                        const newScale = clamp(scale + e.deltaY * 0.001, 0.5, 2);
```

Comment:
```suggestion
                        const newScale = clamp(scale + e.deltaY * 0.001, 0.5, 2);
                        const offset...
```

i think you still need offset.. to me it looks like the scaling is off mostly because of the clamp function, and then the drawImage doesn't have the right offset.X and offset.Y

Reply:
Hmm, so the clamp function is just keeping `scale` between `0.5` and `2`, and immediately after I'm setting `scale` to `newScale`. I don't think `clamp` is the issue, but I agree that the offset needs to be updated too

Reply:
well the scaling is off... i think it might have something to do with the `scale + e.deltaY * 0.001`